### PR TITLE
Milestone : Project #11 Completed [Enumerable Methods - Ruby]

### PR DESCRIPTION
**Pull request by simonpeterdebbarma**
# Completed [Rebuilding Enumerables](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) in partial requirements of the [Microverse Curriculum](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/57421589/). 💯💯💯

---

### Contributors 👨🏼‍💻
- Simon Debbarma

---

##### General Requirements
- [x] Add descriptive README file to your repositories as described here.
- [x] Use meaningful names for your Pull Requests and add a short summary of its content as described here.
- [x] Use descriptive commit messages as described here.
- [x] Do not commit files that are not related to the project. Instead, add them to .gitignore file.
- [x] Do not commit inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history

##### Section
- [x] Do not commit debugging instructions.
- [x] Keep your code clean.

##### Project Specific
- [x] Remember that you should use the Git Flow for the first time. Pay attention to the “Remember to” and “Code review” sections.
- [x] Take a look at Ruby docs about Enumerable class https://ruby-doc.org/core-2.6.5/Enumerable.html
- [x] As your code will be a bit more complex disable Style/CaseEquality cop, learn how to do it here: https://rubydoc.info/gems/rubocop/0.24.1/frames#Disabling_Cops_within_Source_Code

---

# Project Status :

### CODE REVIEW

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    return my_select { |element| element == true }.empty? if !block_given? && arg.nil?
+
+    if block_given?
+      my_each { |n| return true unless yield n }
+    else
+      proc = validate_args(arg)
+      my_each { |n| return true unless proc.call(n) }
+    end
+    false
+  end
```
Comment (whiz25) [inline enumerables.rb]: - [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
let(:array) { Array.new(100) { rand(0...9) } }
expect(array.my_none?(String)).to be false
```
That should return `false` not `true`
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
let(:words) { %w[dog door rod blade] }
expect(words.my_none?(5)).to be words.none?(5)
```
That code block is supposed to return `false` but your method returns `true`

Comment (simonpeterdebbarma) [reply enumerables.rb]: Regarding my_none, I've rewritten it and fixed the first error, but the second one about the 'pattern other than regex or class' seems to work fine for me. 

I cannot figure out how to fix it. Can you please help?


image


image

Comment (simonpeterdebbarma) [reply enumerables.rb]: Hi, could you explain the second requirement- `pattern other than Regex or Class`

When it's comparing an integer to strings, shouldn't it return `true`?

In the terminal, my_none returns the same output as the none method.

image

**Comment by simonpeterdebbarma** [comment]
@whiz25 Hello

So I fixed the first error, but the second one is confusing me. Please re-test the code.

Thanks!

**Comment by simonpeterdebbarma** [comment]
@whiz25

## Fixed! 🛹

They were pretty minor changes to be made.

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+  def my_none?(arg = nil)
+    if block_given?
+      my_each { |n| return false if yield n }
+    else
+      proc = validate_args(arg)
+      my_each { |n| return true unless proc.call(n) }
+    end
+    false
+  end
```
Comment (imhta) [inline enumerables.rb]: There are some are failing in
* [ ] `my_none?` should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))


```bash
rspec ./spec/spec_enumerables_spec.rb:144 # enumerables my_none? returns true if the block never returns true for all elements
rspec ./spec/spec_enumerables_spec.rb:150 # enumerables my_none? when no block or argument is given returns true only if none of the collection members is true
rspec ./spec/spec_enumerables_spec.rb:157 # enumerables my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
rspec ./spec/spec_enumerables_spec.rb:172 # enumerables my_none? when a pattern other than Regex or a class is given returns true only if none of the collection matches the pattern
```

**Comment by simonpeterdebbarma** [comment]
# Thanks guys 😷

**Pull request merged**